metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/RestoreDbtSiblingsIndices.java

```diff
@@ -0,0 +1,165 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.linkedin.metadata.Constants.*;
+
+
+@Slf4j
+@RequiredArgsConstructor
+public class RestoreDbtSiblingsIndices implements BootstrapStep {
+  private static final String VERSION = "0";
+  private static final String UPGRADE_ID = "restore-dbt-siblings-indices";
+  private static final Urn SIBLING_UPGRADE_URN =
+      EntityKeyUtils.convertEntityKeyToUrn(new DataHubUpgradeKey().setId(UPGRADE_ID), Constants.DATA_HUB_UPGRADE_ENTITY_NAME);
+  private static final Integer BATCH_SIZE = 1000;
+  private static final Integer SLEEP_SECONDS = 120;
+
+  private final EntityService _entityService;
+  private final EntityRegistry _entityRegistry;
+
+  @Override
+  public String name() {
+    return this.getClass().getSimpleName();
+  }
+
+  @Nonnull
+  @Override
+  public ExecutionMode getExecutionMode() {
+    return ExecutionMode.ASYNC;
+  }
+
+  @Override
+  public void execute() throws Exception {
+    log.info("Attempting to run RestoreDbtSiblingsIndices upgrade..");
+    log.info(String.format("Waiting %s seconds..", SLEEP_SECONDS));
+
+    // Sleep to ensure deployment process finishes.
+    Thread.sleep(SLEEP_SECONDS * 1000);
+
+    EntityResponse response = _entityService.getEntityV2(
+        Constants.DATA_HUB_UPGRADE_ENTITY_NAME, SIBLING_UPGRADE_URN,
+        Collections.singleton(Constants.DATA_HUB_UPGRADE_REQUEST_ASPECT_NAME)
+    );
+    if (response != null && response.getAspects().containsKey(Constants.DATA_HUB_UPGRADE_REQUEST_ASPECT_NAME)) {
+      DataMap dataMap = response.getAspects().get(Constants.DATA_HUB_UPGRADE_REQUEST_ASPECT_NAME).getValue().data();
+      DataHubUpgradeRequest request = new DataHubUpgradeRequest(dataMap);
+      if (request.hasVersion() && request.getVersion().equals(VERSION)) {
+        log.info("RestoreDbtSiblingsIndices has run before with this version. Skipping");
+        return;
+      }
+    }
+
+    log.info("Bootstrapping sibling aspects");
+
+    final int rowCount = _entityService.listUrns(DATASET_ENTITY_NAME, 0, 10).getTotal();
+
+    log.info("Found {} dataset entities to attempt to bootstrap", rowCount);
+
+    final AspectSpec datasetAspectSpec =
+        _entityRegistry.getEntitySpec(Constants.DATASET_ENTITY_NAME).getAspectSpec(Constants.UPSTREAM_LINEAGE_ASPECT_NAME);
+    final AuditStamp auditStamp = new AuditStamp().setActor(Urn.createFromString(Constants.SYSTEM_ACTOR)).setTime(System.currentTimeMillis());
+
+    final DataHubUpgradeRequest upgradeRequest = new DataHubUpgradeRequest().setTimestampMs(System.currentTimeMillis()).setVersion(VERSION);
+    ingestUpgradeAspect(Constants.DATA_HUB_UPGRADE_REQUEST_ASPECT_NAME, upgradeRequest, auditStamp);
+
+    int indexedCount = 0;
+    while (indexedCount < rowCount) {
+      getAndRestoreUpstreamLineageIndices(indexedCount, auditStamp, datasetAspectSpec);
```

Comment:
Shouldn't we have some try catch here? 

What happens if this method throws an exception? Then we get into a locked state right?